backend/molgenis-emx2-sql/src/main/java/org/molgenis/emx2/sql/SqlSchemaMetadata.java

```diff
@@ -269,19 +268,12 @@ public List<String> getIneritedRolesForUser(String user) {
     final String username = user.trim();
     List<String> result = new ArrayList<>();
     // need elevated privileges, so clear user and run as root
-    // this is not thread safe therefore must be in a transaction
     getDatabase()
-        .tx(
-            tdb -> {
-              String current = tdb.getActiveUser();
-              try {
-                tdb.becomeAdmin(); // elevate privileges
-                result.addAll(
-                    SqlSchemaMetadataExecutor.getInheritedRoleForUser(
-                        ((SqlDatabase) tdb).getJooq(), getName(), username));
-              } finally {
-                tdb.setActiveUser(current); // reset privileges
-              }
+        .getJooqAsAdmin(
+            adminJooq -> {
+              result.addAll(
+                  SqlSchemaMetadataExecutor.getInheritedRoleForUser(
```

Comment:
if this is the only use case then why not wrap in in a dedicated method , instead of exposing getJooqAsAdmin to package

Reply:
I had that initially. My motivation is to prevent myself reinventing the wheel later, if that makes sense.

Reply:
So I'd rather keep it as is.

Reply:
these's other things to consider but a guess it's a tradeoff